go-controller/pkg/allocator/pod/pod_annotation.go

```diff
@@ -400,6 +400,16 @@ func allocatePodAnnotationWithRollback(
 		hasIPAMClaim = ipamClaim != nil && len(ipamClaim.Status.IPs) > 0
 	}
 
+	defer func() {
+		if ipamClaim == nil || claimsReconciler == nil {
+			return
+		}
+		updatedClaim := claimsReconciler.UpdateIPAMClaimStatus(ipamClaim, podAnnotation, pod.Name, err)
```

Comment:
Something I just thought of...
With this change any pod that tries to miss-behave will be able to wipe the ipamclaims status and move it to failed state, is that correct?
If there is an IPAMClaims that exists already and I create a random pod that references it to mess with it will we loose the status? I am not sure if we can do much about it but let's get a common understanding on what would happen and if it is a real issue.